autometa/common/external/hmmer.py

```diff
@@ -205,64 +222,64 @@ def filter_markers(infpath, outfpath, cutoffs, orfs=None, force=False):
+    df["cleaned_sacc"] = df["sacc"].map(lambda acc: acc.split(".")[0])
+    dff = pd.read_csv(cutoffs, sep="\t", index_col="accession")
+    mdf = pd.merge(df, dff, how="left", left_on="cleaned_sacc", right_on="accession")
+    mdf = mdf[mdf["score"] >= mdf["cutoff"]]
     if mdf.empty:
-        raise AssertionError(f'No markers in {infpath} pass cutoff thresholds')
-    cols = ['orf', 'sacc', 'sname', 'score', 'cutoff']
+        raise AssertionError(f"No markers in {infpath} pass cutoff thresholds")
+    cols = ["orf", "sacc", "sname", "score", "cutoff"]
     mdf = mdf[cols]
     translations = prodigal.contigs_from_headers(orfs)
-    def translater(x): return translations.get(x, x.rsplit('_', 1)[0])
-    mdf['contig'] = mdf['orf'].map(translater)
-    mdf.set_index('contig', inplace=True)
-    mdf.to_csv(outfpath, sep='\t', index=True, header=True)
+
+    def translater(x):
+        return translations.get(x, x.rsplit("_", 1)[0])
+
+    mdf["contig"] = mdf["orf"].map(translater)
+    mdf.set_index("contig", inplace=True)
+    mdf.to_csv(outfpath, sep="\t", index=True, header=True)
     return outfpath
 
 
 def main():
     import argparse
     import logging as logger
+
     logger.basicConfig(
-        format='%(asctime)s : %(name)s : %(levelname)s : %(message)s',
-        datefmt='%m/%d/%Y %I:%M:%S %p')
+        format="%(asctime)s : %(name)s : %(levelname)s : %(message)s",
+        datefmt="%m/%d/%Y %I:%M:%S %p",
+        level=logger.DEBUG,
+    )
     parser = argparse.ArgumentParser(
-        description='Retrieves markers with provided input assembly')
-    parser.add_argument('orfs', help='</path/to/assembly.orfs.faa>')
-    parser.add_argument('hmmdb', help='</path/to/hmmpressed/hmmdb>')
-    parser.add_argument('cutoffs', help='</path/to/hmm/cutoffs.tsv>')
-    parser.add_argument('hmmscan', help='</path/to/hmmscan.out>')
-    parser.add_argument('markers', help='</path/to/markers.tsv>')
-    parser.add_argument('--log', help='</path/to/parallel.log>')
-    parser.add_argument('--force', help="force overwrite of out filepath",
-                        action='store_true')
-    parser.add_argument('--cpus', help='num cpus to use', default=0, type=int)
+        description="Retrieves markers with provided input assembly"
+    )
+    parser.add_argument("orfs", help="</path/to/assembly.orfs.faa>")
+    parser.add_argument("hmmdb", help="</path/to/hmmpressed/hmmdb>")
+    parser.add_argument("cutoffs", help="</path/to/hmm/cutoffs.tsv>")
+    parser.add_argument("hmmscan", help="</path/to/hmmscan.out>")
+    parser.add_argument("markers", help="</path/to/markers.tsv>")
     parser.add_argument(
-        '--parallel', help="enable GNU parallel", action='store_true')
-    parser.add_argument('--verbose', help="add verbosity", action='store_true')
+        "--force", help="force overwrite of out filepath", action="store_true"
+    )
+    parser.add_argument("--cpus", help="num cpus to use", default=0, type=int)
```

Comment:
Specifying `cpus=0` will tell GNU parallel to use as many as possible. This is why I've set the default to 0 rather than `mp.cpu_count()`

Reply:
I was unable to find what specifying `cpus=0` would do for hmmscan, but I found this on the hmmer manual:

> If you specify--cpu 0, the program will run in serial-only mode, with no threads. This might be useful if you suspect something is awry with the threaded parallel implementation

Correct me if I am wrong and I am sure you must have double-checked this but is the application of `cpus=0` different for different functions then?

Reply:
Yes, it seems `cpus=0` causes different behavior between GNU parallel and hmmer. This may be we were seeing conflicting performance results. We should update this to handle the cpu input appropriately. Nice find! 🔍 🐎

Reply:
So it that in GNU parallel `cpus=0` uses all cpus. Sorry I still was not able to find some documentation that states this behaviour. Is it possible to post it here?

Reply:
From `man parallel`

```shell
       --jobs N
       -j N
       --max-procs N
       -P N     Number of jobslots on each machine. Run up to N jobs in parallel.  0 means as many as possible.
                Default is 100% which will run one job per CPU core on each machine.

                If --semaphore is set, the default is 1 thus making a mutex.